### ISSUE TITLE
Add option to quickly sign into google from browser (Firefox, Chrome, etc)

### ISSUE THREAD

**Issue by flleeppyy** [ytmdesktop/ytmdesktop#626]
- [x] I understand that **YTMDesktop have NO affiliation with Google or YouTube**.
- [x] I verified that there is no open issue for the same subject.

**Is your feature request related to a problem? Please describe.**
Signing into google is a bit of a pain for me, since my password is long, and I want to be able to click a button and have it go to my browser, authenticate me with google and then go back to Youtube Music without having to enter a password

**Describe the solution you'd like**
Add a button on the Google account sign in page to sign in through the browser if the user is already logged in there, so they don't have to enter their password.

I would love to implement this myself but this isn't my area of expertise (yet)

**Comment by NovusTheory** [comment]
Unfortunately with the way browser's work I can't see this being possible. There would be no way for the app to read cookies from Google to use in the app. It would require more hoops, such as a browser extension, to make this possible which I'm sure users wouldn't want to do.

**Comment by flleeppyy** [comment]
Well, from what ive seen applications do, I was thinking that the YT client hosts a web server and opens a link in the web browser, that when authorized through google it would go to that url, and sign in for the user

**Comment by NovusTheory** [comment]
OAuth wouldn't work in this instance which is what you're talking about

**Comment by flleeppyy** [comment]
Well... shit.

Oh well.

**Comment by mingjun97** [comment]
Should be a good way to fix the malfunction part for some 2FA accounts. But we don't know how to implement it. YT client could do this since they are using sort of API and they are able to pass the token through request, while our YTMDesktop is just a wrapper of YTM web. 

Any attempts is welcome.TBH, I don't think we can do it without steal token from browsers. Lol

**Comment by NovusTheory** [comment]
I looked at Google's sign in system from YTM a while ago and while it is OAuth they have a privileged OAuth and we're not able to change the redirect url to for example one that this app uses. So I can't see this being do-able without hassle like I mentioned above.

**Comment by rnett** [comment]
That's unfortunate, for me entering sign in info and especially security keys into a 3rd party application is unacceptable, security wise.  I doubt I'm the only person with that limitation.  A browser extension, even if inconvenient, would work fine for me, and I suspect most people with these security concerns wouldn't mind having to jump through a few hoops.  Of course that doesn't include dev time considerations.

**Comment by rnett** [comment]
Actually, this is the perfect scenario to use an app passwords, except that does not seem to work when I try it.

**Comment by Alipoodle** [comment]
I'm moving to a new workplace soon where I will be given a Security Key, so I plan to look more into issues with Signing in more later on.
TL;DR App-password, 2FA or Security key are your options to really use regarding sign in and feel like we're not stealing your information, which is why this is Open Source in the first place for people to build their own version and view the code.

I doubt that "Grabbing a Session token" (hypothetical method) sounds any better than just signing in directly to the app.

**Comment by Alipoodle** [comment]
I'm marking this as Stale and not possible.
As a Reminder we are not YTM and Google provides no access to a OAuth style way of authenticating.

Ultimately if you do want to thing of things regarding using Security Keys in a 3rd Party apps, you need to remember how secure they are 😅 and that things are done on a per-site basis and the combination of the key's contents changes every second/millisecond.

Obviously we're open source so you can view all the source code, and even build it yourself if you feel that there's a risk in the builds which we are providing to you in the Releases tab.
We're obviously directing you straight to Google for the Login, and the only thing we add to the page ("Off the rails") is a Back button to return back to YTM if you can't sign in.

**Comment by tgr** [comment]
You can listen on some port, use `127.0.0.1:<your port>` as the redirect URI and capture the OAuth token there. [See docs.](https://developers.google.com/identity/protocols/oauth2/native-app#redirect-uri_loopback)

You could also set up a small web application, use it as the return URL for Google OAuth, and then just print the access token and tell the user to copy it to your app. This is essentially the same as the now-removed [manual copy-paste](https://developers.google.com/identity/protocols/oauth2/native-app#manual-copypaste) method, though, so presumably Google doesn't like it.

There's also a [more complicated OAuth flow](https://developers.google.com/identity/protocols/oauth2/limited-input-device) for apps which cannot interact with a browser (e.g. smart TV apps).

**Comment by Alipoodle** [comment]
OAUTH TOKEN IS NOT POSSIBLE FOR YOUTUBE MUSIC as it is a direct service which uses your full Google account and not just the email or information from you Google Account

THIS IS NOT POSSIBLE with the OAuth Flow.

**Comment by tgr** [comment]
Yeah, sorry, I misunderstood the [earlier comment](#issuecomment-780061682) about OAuth limitation.

It seems like even the Youtube API doesn't really support interacting with autogenerated playlists (the feature Youtube Music grew out of), and YTM has no public API whatsoever, so apparently music playlists is not something Google wants developers to work with :(
